src/client/eppo-precomputed-client.ts

```diff
@@ -51,32 +67,38 @@ export default class EppoPrecomputedClient {
   private precomputedFlagsRequestParameters?: PrecomputedFlagsRequestParameters;
   private subjectKey?: string;
   private subjectAttributes?: Attributes;
+  private flagKeySalt = '';
+  private precomputedFlagStore: IConfigurationStore<PrecomputedFlag>;
+  private isObfuscated: boolean;
 
-  constructor(
-    private precomputedFlagStore: IConfigurationStore<PrecomputedFlag>,
-    private isObfuscated = false,
-  ) {}
+  constructor(options: EppoPrecomputedClientOptions) {
+    this.precomputedFlagStore = options.precomputedFlagStore;
+    this.isObfuscated = options.isObfuscated ?? true; // Default to true if not provided
+  }
 
-  public setPrecomputedFlagsRequestParameters(
-    precomputedFlagsRequestParameters: PrecomputedFlagsRequestParameters,
-  ) {
-    this.precomputedFlagsRequestParameters = precomputedFlagsRequestParameters;
+  public setIsObfuscated(isObfuscated: boolean) {
+    this.isObfuscated = isObfuscated;
   }
 
-  public setSubjectAndPrecomputedFlagsRequestParameters(
-    precomputedFlagsRequestParameters: PrecomputedFlagsRequestParameters,
-  ) {
-    this.setPrecomputedFlagsRequestParameters(precomputedFlagsRequestParameters);
-    this.subjectKey = precomputedFlagsRequestParameters.precompute.subjectKey;
-    this.subjectAttributes = precomputedFlagsRequestParameters.precompute.subjectAttributes;
+  private setFlagKeySalt(salt: string) {
+    this.flagKeySalt = salt;
   }
 
-  public setPrecomputedFlagStore(precomputedFlagStore: IConfigurationStore<PrecomputedFlag>) {
-    this.precomputedFlagStore = precomputedFlagStore;
+  private setPrecomputedFlagsRequestParameters(parameters: PrecomputedFlagsRequestParameters) {
+    this.precomputedFlagsRequestParameters = parameters;
   }
 
-  public setIsObfuscated(isObfuscated: boolean) {
-    this.isObfuscated = isObfuscated;
+  private setSubjectData(subjectKey: string, subjectAttributes: Attributes) {
+    this.subjectKey = subjectKey;
+    this.subjectAttributes = subjectAttributes;
+  }
+
+  // Convenience method that combines setters we need to make assignments work
+  public setSubjectAndPrecomputedFlagsRequestParameters(
+    parameters: PrecomputedFlagsRequestParameters,
+  ) {
+    this.setPrecomputedFlagsRequestParameters(parameters);
+    this.setSubjectData(parameters.precompute.subjectKey, parameters.precompute.subjectAttributes);
```

Comment:
This should not be public.

It also seems to be doing a bad thing — allows reinterpreting precomputed configuration from another subject. It also does not reset the poller, so it would continue polling with the previous subject key/attributes

Reply:
made not public.